### PR TITLE
feat: WinUI `TimePicker`

### LINKED ISSUE #5553: `TimePicker` from WinUI
Adapt the WinUI's TimePicker to Uno

- [ ] Add native TimePicker ui test (https://github.com/unoplatform/uno/pull/5720)

## Related issues

- [iOS] TimePicker Property MinuteIncrements is not working #3540
- TimePicker does not use WinUI template #5990
- [iOS][Android] TimePicker, HasTimeStates are not working #4123
- [Gallery][Android]<TimePicker> TimePicker features are not working. #5366
- https://github.com/unoplatform/uno/issues/6658

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #5553

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

We don't have a `TimePicker` on non-mobile targets.

## What is the new behavior?

We have a `TimePicker` on non-mobile targets 🥳 !

### <samp>🤖 Generated by Copilot at 01b3ed1</samp>

This pull request adds the `TimePicker` and `TimePickerFlyout` controls to the Uno Platform, which are used to select a time value. The pull request also adds the `InputKeyboardSource` class, which provides keyboard state information, and updates the `AutomationHelper` and `PlatformHelpers` classes, which provide automation and platform-specific features. The pull request also uses the new C# 10 file-scoped namespace syntax and the nullable reference types feature in some files.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

**Comment by MartinZikmund** [comment]
Current issue:

When PM, the 12 hour clock on `Calendar` does not work. `Period` setter is incorrect as it does not actually store the value anywhere.

**Comment by MartinZikmund** [comment]
Issues reported by Jerome:

1. Initial PM switches to AM

20240110-timepicker

2. The flyout is initially positioned incorrectly, then snaps into position

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/MonochromaticOverlayPresenter/MonochromaticOverlayPresenter.cs [hunk 1]
```diff
@@ -45,68 +45,68 @@ private void InvalidateBrush()
...
 		{
-			if (SourceElement is { } targetElement)
-			{
-				var newColor = ReplacementColor;
-				if (_replacementColor != newColor)
-				{
-					_replacementColor = newColor;
-					_effectFactory = null;
-				}
+			//if (SourceElement is { } targetElement)
```
Comment (jeromelaban) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/MonochromaticOverlayPresenter/MonochromaticOverlayPresenter.cs:48]: @MartinZikmund we did not need this at all ?

Comment (MartinZikmund) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/MonochromaticOverlayPresenter/MonochromaticOverlayPresenter.cs:48]: It didn't work actually, and threw an exception which was masked, so I instead adjusted the styles so it is not needed. But will ad a runtime check instead to keep the code in

**Pull request merged**